metals/src/main/resources/MetalsSbtBsp.scala

```diff
@@ -0,0 +1,64 @@
+package metals
```

Comment:
For sure it would be good to have it as a separate plugin, we did release an sbt plugin before and it should not be too much of an additional effort.

Reply:
Sounds good. That's what I'll plan on doing. I'll probably due a separate pr including the plugin, and then we can merge it in and dogfood it to this pr so it's available.

Reply:
My plan for this is to after it's to a point where we like it, send in a separate pr to add this to the build and to the release. Then after it's merged in, just use the snapshot that is produced and dogfeed it to this pr. Does that sound alright?

Reply:
Ach, duh, yea that should actually work fine. I'll work on migrating this over as a plugin in this pr.

Reply:
Playing around with this a bit, should we include both plugins in the `metals.sbt` file? I'm thinking through the best way to do this, and I guess we could do it a few different ways:

1. Two different files (Probably not)
2. One file that includes both the `sbt-metals` and `sbt-bloop` plugin
3. One file that dynamically gets the `sbt-metals` or `sbt-bloop` plugin placed in

I'm leaning a bit towards 2 just for the easiness of it, however, then we also have to worry about users that potentially use the `bloopSbtAlreadyInstalled` user config. However, we could just change the logic in that case to only include `sbt-metals` then.


What do you think?

Reply:
I think it would be best if we used 3. Adding the sbt plugin will add semanticdb to the build definition, which will not be neccessary if we used Bloop.  Let's only add the new metals plugin is anyone invokes generateBspConfig, especially now as we still do not recommend using sbt.

Reply:
Alright, sounds good. I went with 3 and it seems to be working fine 👍

Reply:
Alright, sounds good. I went with 3 and it seems to be working fine 👍